src/app/ReadHandler.cpp

```diff
@@ -25,14 +25,17 @@
 #include <app/AppBuildConfig.h>
 #include <app/InteractionModelEngine.h>
 #include <app/MessageDef/EventPath.h>
+#include <app/MessageDef/SubscribeRequest.h>
+#include <app/MessageDef/SubscribeResponse.h>
 #include <app/ReadHandler.h>
 #include <app/reporting/Engine.h>
+#include <lib/support/RandUtils.h>
 #include <protocols/secure_channel/StatusReport.h>
 
 namespace chip {
 namespace app {
 CHIP_ERROR ReadHandler::Init(Messaging::ExchangeManager * apExchangeMgr, InteractionModelDelegate * apDelegate,
-                             Messaging::ExchangeContext * apExchangeContext)
+                             Messaging::ExchangeContext * apExchangeContext, InteractionType aInteractionType)
```

Comment:
Can't comment in the right section, but just a note that right now, the ReadHandler is generating new exchanges for each report messages, even if all the reports are chunked as part of the same Action. This needs fixing, but can happen later.

Reply:
During subscription establishenment procedure, the initial report is using same exchange context. During post establishment, the report is using new exchange context.